src/main/java/team/serenity/logic/commands/question/EditQnCommand.java

```diff
@@ -0,0 +1,198 @@
+import team.serenity.model.group.GroupName;
+import team.serenity.model.group.lesson.LessonName;
+import team.serenity.model.group.question.Description;
+import team.serenity.model.group.question.Question;
+
+/**
+ * Edits the description of an existing question.
+ */
+public class EditQnCommand extends Command {
+
+    public static final String COMMAND_WORD = "editqn";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the question identified "
+            + "by the index number used in the displayed list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_GRP + "GROUP_NAME] "
+            + "[" + PREFIX_LSN + "LESSON_NAME] "
+            + "[" + PREFIX_QN + "QUESTION]\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_QN + "Can you repeat the deadlines for all submissions?";
+
+    public static final String MESSAGE_EDIT_QUESTION_SUCCESS = "Edited Question: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "Question is not edited.\n"
+        + "At least one field needs to be change from the original question.";
+    public static final String MESSAGE_GROUP_NOT_FOUND_FORMAT = "This group \"%s\" does not exists.";
+    public static final String MESSAGE_LESSON_NOT_FOUND_FORMAT = "This lesson \"%s\" does not exists.";
+    public static final String MESSAGE_DUPLICATE_QUESTION = "This question already exists in the list.";
+
+
+    private final Index index;
+    private final EditQuestionDescriptor editQuestionDescriptor;
+
+    /**
+     * @param index                  of the question in the filtered question list to edit
+     * @param editQuestionDescriptor details to edit the activity with
+     */
+    public EditQnCommand(Index index, EditQuestionDescriptor editQuestionDescriptor) {
+        requireAllNonNull(index, editQuestionDescriptor);
+        this.index = index;
+        this.editQuestionDescriptor = new EditQuestionDescriptor(editQuestionDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        List<Question> lastShownList = model.getFilteredQuestionList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(MESSAGE_INVALID_QUESTION_DISPLAYED_INDEX);
+        }
+
+        Question questionToEdit = lastShownList.get(index.getZeroBased());
+        Question editedQuestion = createEditedQuestion(questionToEdit, editQuestionDescriptor);
+
+        // Checks if the edited question is the same as question to edit
+        if (questionToEdit.equals(editedQuestion)) {
+            throw new CommandException(MESSAGE_NOT_EDITED);
+        }
+
+        // Checks if the edited question exists in the question manager
+        if (!questionToEdit.equals(editedQuestion) && model.hasQuestion(editedQuestion)) {
```

Comment:
Since equality is already checked in  `if (questionToEdit.equals(editedQuestion))` , is there a need to check for `!questionToEdit.equals(editedQuestion)` ? 

Likewise for the other 2 if blocks below => Since the questions are already not equal, do we need to check the groupName and lessonName equality?

Reply:
Realised it is already accounted for in the previous clause, will remove it!!

For the 2 `if` blocks below, the other attributes of `questionToEdit` and `editedQuestion` could be different and not be caught by the `if (questionToEdit.equals(editedQuestion))` block. The 2 checks are mainly to ensure that the `editedQuestion` group and lesson names are valid, assuming that they were edited.